config/locales/en.yml

```diff
@@ -263,11 +263,13 @@ en:
     edit:
       heading: Associate team to
   user_journey:
+    errors:
+      select_option: Must select an option
     title: Dashboard
     component_name:
       MSA: Matching Service Adapter
       VSP: Verify Service Provider
-      SP: Service Provider
+      'service provider': Service Provider
```

Comment:
why this can't stay as `SP`?

Reply:
The keyword `SP` is never used anywhere. And the method `display_component` returns rather `MSA`, `VSP` or `service provider`.  The method is used in two ways, one to display one of the 3 outcomes directly or it is used to located the keyword in the yml file to display different text.

Reply:
SP should be used as the key (or `service_provider`)... having the value as a key `'service provider'` defeats the purpose of the locale file .. and there should be no spaces in the key names ideally...

Reply:
The method doesn't actually ever print directly, it is used in interpolation, so what about this scenario where the method is used to pass in the constant directly?: `t('user_journey.add_new_signing_key_warning', component: display_component(@certificate.component))`

Reply:
should then the `display_component()` return the interpolated value instead? maybe?

Reply:
But thats what its technically already doing? because the constant it returns is the value being used

Reply:
I think the `display_component` method is doing two quite different things. Given its name, it should return the display value for the component type, i.e. it should do `t()` and return the display value... for interpolation we should have a different method which returns the constant (`MSA`, `VSP` or `SP`) ... let's leave it for another PR though...

Reply:
Yeah i understand, would be better to raise it as another PR